app/pipeline/pdf_extractor.py

```
@@ -0,0 +1,168 @@
+            if page_text.strip():
+                text_parts.append(page_text)
+        
+        doc.close()
+
+        # 페이지별 텍스트를 줄바꿈으로 결합
+        full_text = "\n".join(text_parts)
+        
+        # 후처리: 불필요한 공백 정리
+        full_text = _clean_pdf_text(full_text)
+
+        return full_text
```

Ruff (0.14.7)

Consider moving this statement to an `else` block

(TRY300)